internal/authentication/authentication.go

```diff
@@ -19,20 +18,19 @@ type AuthToken = string
 type Authenticator[AuthCredientals any] interface {
 	CreateToken(AuthCredientals) (AuthToken, error)
 	RevokeToken(AuthToken) error
-	CheckToken(AuthToken) bool
+	CheckToken(AuthToken) (string, error)
```

Comment:
What's does this return type meen? Please add a doc-comment explaining. (And possibly a newtype'd string)

Reply:
Documented in https://github.com/gpuctl/gpuctl/pull/140/commits/2e2016f2d2c26c2c6af68140091bc2f94a2f8d8a
Made a type-alias as prep for newtyping, but I don't feel confident enough with the semantics of the back-end or best Go-practice to actually go ahead refactor the back-end to use a newtype (i.e. work out where all conversions ought to be).